### PR TITLE
Renaming

### PR DESCRIPTION
**Pull request by FarnazH**
## Steps

- [x] Write a good description of what the PR does.
- [x] Update documentation
- [x] Rebase onto master

## Description
Rename some of the modules and functions to be more intuitive. I tried to catch every appearance of the module/function, but I might have missed something.

## Type of Changes
|   | Type |
| ------------- | ------------- |
| ✓  | :hammer: Refactoring  |

### CODE REVIEW

**Comment by FarnazH** [comment]
[Make names of modules/functions more intuitive](https://trello.com/c/4fgG6hw0/80-make-names-of-modules-functions-more-intuitive)

**Comment by FarnazH** [comment]
[Make names of modules/functions more intuitive](https://trello.com/c/4fgG6hw0/80-make-names-of-modules-functions-more-intuitive)

File: gbasis/evals/basis_eval.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import numpy as np
 
 
-class Eval(BaseOneIndex):
+class BasisEval(BaseOneIndex):
```
Comment (wilhadams) [inline gbasis/evals/basis_eval.py:8]: This is so much better

**Review by wilhadams — COMMENTED** [review]
It might make sense to rename `evaluate_deriv_density` to `evaluate_density_deriv` in order to align it with `evaluate_basis_deriv` syntax-wise.

File: gbasis/evals/basis_eval.py (hunk 1 above)
Comment (FarnazH) [reply gbasis/evals/basis_eval.py:8]: Thanks for your comment @wilhadams. See 26ae9aa

**Review by kimt33 — COMMENTED** [review]
I think these changes are fine. The main issue at this point is that there are other PR's that may break if this PR gets merged. And I'm not sure if they'll get modified to accommodate this change.

I'm sorry that it took so long for me to look at this PR. This wouldn't be a problem if the PR got merged earlier. This is also pretty tedious work and I wouldn't want you to redo all this stuff.

I informally approve this PR but won't officially approve it because it might cause a lot of headache if it gets accidentally meged in the correct order. Once the approved PR's get merged, hopefully this PR only need minor modifications to merge. Otherwise, we can just leave the PR up for someone else to make these changes.

**Pull request closed without merging**